ambry-tools/src/main/java/com.github.ambry.tools/admin/DumpData.java

```diff
@@ -731,6 +765,10 @@ public int compare(Pair<Long, Long> o1, Pair<Long, Long> o2) {
       });
     }
 
+    /**
+     * Adds and merges the interval if need be
+     * @param newInterval the new interval that needs to be added
+     */
```

Comment:
Nope. as of now MergedInterval class doesn't handle this case. We can fix it going forward if we see such cases.